src/sink.rs

```diff
@@ -446,4 +446,24 @@ mod tests {
             assert_eq!(queue_rx.next(), src.next());
         }
     }
+
+    #[test]
+    fn test_channel_volume() {
+        use crate::source::ChannelVolume;
+
+        let (sink, mut queue_rx) = Sink::new_idle();
+
+        let v = vec![100.0f32];
+        let input = SamplesBuffer::new(1, 48000, v.clone());
+
+        // High rate to avoid immediate control.
```

Comment:
Can you explain this comment?

Reply:
Honestly, no idea. I'll be spending the day today going through this throughly to actually try to understand what I did.

Reply:
I'm going to remove the test here. I don't think it's actually helpful to have there